unit_test/test_spdm_requester/encap_request.c

```diff
@@ -93,12 +93,9 @@ libspdm_return_t libspdm_requester_encap_request_test_receive_message(
                             (uint8_t)(0xFF));
 
             digest = (void *)(spdm_response + 1);
-            /*send all eight certchains digest
-             * but only No.7 is right*/
-            digest += libspdm_get_hash_size(m_libspdm_use_hash_algo) * (SPDM_MAX_SLOT_COUNT - 2);
             libspdm_hash_all(m_libspdm_use_hash_algo, m_libspdm_local_certificate_chain,
                              LIBSPDM_MAX_MESSAGE_BUFFER_SIZE, &digest[0]);
-            spdm_response->header.param2 |= (0xFF << 0);
+            spdm_response->header.param2 |= (0x01 << 0);
```

Comment:
why we need change this? Is it related to raw public key provisioning?

Reply:
This is changed because both of

provisioned peer certificate chain in local context is gone
the way that libspdm_verify_peer_digests() is changed
Previously the way of testing

send all eight certchains digest but only No.7 is right

would be succeeded because we were doing it in the way that to see if there is digest of one slot in retrieved digests data matches the hash of provisioned peer chetificate chain in local context.

Since provisioned peer certificate chain in local context is gone,
and we are caching peer certificate chain data/hash slot by slot in GET_CERTIFICATE process,
the way to check peer digests in libspdm_verify_peer_digests() is also changed to check the hash value of every slot according to given slot_mask.

And due to that, send all eight certchains digest but only No.7 is right way of testing won't be succeeded any more.
Now pass condition for GET_DIGESTS should be: hash value for every slot is correct against the given slot_mask

Reply:
I will remain the way to generate digest in test routine as the way it is right now (only send correct digest value of one slot for most cases), because the previous way of digest generation
> send all eight certchains digest but only No.7 is right

is no longer needed, since `libspdm_verify_peer_digests()` will be removed at all.
Refer to https://github.com/DMTF/libspdm/pull/1556#discussion_r1063003386